core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -333,6 +347,122 @@ public long nextFetchOffset() {
         }
     }
 
+    void updateOffsetsOnLsoMovement() {
+        long logStartOffset = offsetForEarliestTimestamp();
+        lock.writeLock().lock();
+        try {
+            if (logStartOffset > startOffset) {
```

Comment:
If we have called this method that means an exception occurred and if this condition is false then what we want to do, just ignore?

Reply:
yes, we can remove it.